controllers/grid/settings/user/UserGridHandler.inc.php

```diff
@@ -44,8 +46,13 @@ function __construct() {
 	 * @copydoc PKPHandler::authorize()
 	 */
 	function authorize($request, &$args, $roleAssignments) {
-		import('lib.pkp.classes.security.authorization.ContextAccessPolicy');
-		$this->addPolicy(new ContextAccessPolicy($request, $roleAssignments));
+
+		// Allow site admin to access this grid without a context
+		$currentUser = $request->getUser();
+		if ($request->getContext() || !$currentUser || !$currentUser->hasRole(ROLE_ID_SITE_ADMIN, CONTEXT_ID_NONE)) {
```

Comment:
shouldn't currentUser exist for ContextAccessPolicy?

Reply:
Yeah, it will fail if no `$currentUser` exists. 

So the goal of this block is: is this a context request and is the user a site admin? If not, then force them to pass a context policy. If this isn't a context _and_ the user is a site admin, they don't have to pass the context policy.

Is that right or have I messed the logic up here?

Reply:
I was just wandering if the check !$currentUser is necessary. But now I see: if !$currentuser the ContextAccessPolicy will fail, else the function hasRole can be executed properly (without error) because the currentUser exists. :+1: :+1: :+1: